core/src/main/java/org/elasticsearch/rest/action/admin/indices/analyze/RestAnalyzeAction.java

```diff
@@ -77,17 +79,17 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
         analyzeRequest.text(texts);
         analyzeRequest.analyzer(request.param("analyzer"));
         analyzeRequest.field(request.param("field"));
-        analyzeRequest.tokenizer(request.param("tokenizer"));
-        analyzeRequest.tokenFilters(request.paramAsStringArray("filter", request.paramAsStringArray("token_filter", analyzeRequest.tokenFilters())));
-        analyzeRequest.charFilters(request.paramAsStringArray("char_filter", analyzeRequest.charFilters()));
         analyzeRequest.explain(request.paramAsBoolean("explain", false));
         analyzeRequest.attributes(request.paramAsStringArray("attributes", analyzeRequest.attributes()));
+        parseTokenizer(analyzeRequest, request.param("tokenizer"));
+        parseTokenFilters(analyzeRequest, request.paramAsStringArray("filter", request.paramAsStringArray("token_filter", Strings.EMPTY_ARRAY)));
+        parseCharFilters(analyzeRequest, request.paramAsStringArray("char_filter", Strings.EMPTY_ARRAY));
```

Comment:
do we really need to support providing json configurations of tokenizers, token filters and char filters it query string parameters?